src/site/twitch.tv/modules/settings/SettingsMenuButton.vue

```diff
@@ -11,37 +11,103 @@
 </template>
 
 <script setup lang="ts">
-import { ref } from "vue";
+import { reactive, ref, toRaw, watch, watchEffect } from "vue";
+import { HookedInstance } from "@/common/ReactHooks";
+import { defineFunctionHook } from "@/common/Reflection";
+import { useConfig } from "@/composable/useSettings";
 import useUpdater from "@/composable/useUpdater";
 import Logo7TV from "@/assets/svg/logos/Logo7TV.vue";
 
+const updater = useUpdater();
+
 const emit = defineEmits<{
 	(event: "toggle"): void;
 }>();
 
-const updater = useUpdater();
+const props = defineProps<{
+	inst?: HookedInstance<Twitch.TopNavBarComponent>;
+}>();
+
+const hideBitsButton = useConfig<boolean>("layout.hide_bits_buttons");
+const hideSiteNotificationsButton = useConfig<boolean>("layout.hide_site_notifications_button");
+const hidePrimeCrown = useConfig<boolean>("layout.hide_prime_offers");
+const hideTurboButton = useConfig<boolean>("layout.hide_turbo_button");
+const hideWhispersButton = useConfig<boolean>("layout.hide_whispers_button");
+
+const menuElements = reactive({
+	renderBitsButton: hideBitsButton,
+	renderOnsiteNotificatons: hideSiteNotificationsButton,
+	renderTwitchPrimeCrown: hidePrimeCrown,
+	renderTurboButton: hideTurboButton,
+	renderWhispers: hideWhispersButton,
+});
+
+// force update top nav when user changes settings
+watch(menuElements, (renderElements) => {
+	rerenderTopNav(renderElements);
+});
+
+// TODO: define proper types here
+function rerenderTopNav(els) {
```

Comment:
I don't really understand this at all, why are we trying to redefine the hook multiple times after each settings change?

Reply:
Shouldn't this function just call `forceUpdate()`?